### DISCUSSION TITLE
Strimzi Kafka Operator upgrade plan from 0.34.0 to 0.43.0

### DISCUSSION THREAD

**Discussion by bharat15-ml** [strimzi/strimzi-kafka-operator#10838] [General]
Hi Team,

I am trying to upgrade Strimzi Kafka operator from version 0.34.0 to 0.43.0. It seems operator version 0.43.0 doesn't support downgraded Kafka cluster deployment which is in my case of version 3.4.0 which deployed with operator version 0.34.0

Pls suggest upgrade plan in this case. Any help is well appreciated.

1. Current deployment - 
Kafka Operator 0.34.0
Kafka Cluster Version - 3.4.0

2. Expected Upgrade plan - 
Kafka Operator 0.43.0
Kafka Cluster Version - 3.8.0

**Comment by scholzj** [comment]
I'm not sure I follow the question or the issue. 

The upgrade process is described in our docs. But given how old version you use right now, you should definitely first try to test it in some test environment.

**Comment by katheris** [comment]
@bharat15-ml once you have upgraded the Strimzi operator you can then change the `spec.kafka.version` in the Kafka CR to 3.8.0 and Strimzi should update the Kafka cluster. Until you update the Kafka CR you will get an [unsupported Kafka version error.](https://strimzi.io/docs/operators/latest/deploying#con-upgrade-cluster-operator-unsupported-kafka-str) As Jakub said you are jumping many versions so definitely test it first.

If you would rather do a more incremental upgrade you could first upgrade Strimzi operator to 0.37 which does support Kafka 3.4.0, then upgrade Kafka to 3.5.1, then you can upgrade Strimzi again etc until you get to the versions you want. You can see all the supported versions on the [website](https://strimzi.io/downloads/) to work out the upgrade path with the least jumps.

**Reply by bharat15-ml** [reply]
Thanks, @scholzj and @katheris. I have tried your suggested points and can successfully upgrade the Kafka operator and Kafka cluster versions.
